crates/swc_html_minifier/src/lib.rs

```diff
@@ -470,351 +471,78 @@ impl Minifier<'_> {
-                | (Namespace::SVG, "tspan", "font-stretch", "normal")
-                | (Namespace::SVG, "tspan", "font-style", "normal")
-                | (Namespace::SVG, "tspan", "font-variant", "normal")
-                | (Namespace::SVG, "tspan", "font-weight", "normal")
-                | (Namespace::SVG, "use", "opacity", "1")
-                | (Namespace::SVG, "view", "zoomAndPan", "magnify")
-                | (
-                    Namespace::MATHML,
-                    "math",
-                    "xmlns",
-                    "http://www.w3.org/1998/math/mathml"
-                )
-                | (
-                    Namespace::MATHML,
-                    "math",
-                    "xlink",
-                    "http://www.w3.org/1999/xlink"
+        match namespace {
+            Namespace::HTML | Namespace::SVG => {
+                // Legacy attributes, not in spec
+                if tag_name == "script" {
+                    match attribute_name {
+                        "type" => {
+                            match &*attribute_value.trim().to_ascii_lowercase() {
+                                // Legacy JavaScript MIME types
+                                "application/javascript"
+                                | "application/ecmascript"
+                                | "application/x-ecmascript"
+                                | "application/x-javascript"
+                                | "text/ecmascript"
+                                | "text/javascript1.0"
+                                | "text/javascript1.1"
+                                | "text/javascript1.2"
+                                | "text/javascript1.3"
+                                | "text/javascript1.4"
+                                | "text/javascript1.5"
+                                | "text/jscript"
+                                | "text/livescript"
+                                | "text/x-ecmascript"
+                                | "text/x-javascript" => return true,
+                                _ => {}
+                            }
+                        }
+                        "language" => match &*attribute_value.trim().to_ascii_lowercase() {
+                            "javascript" | "javascript1.2" | "javascript1.3" | "javascript1.4"
+                            | "javascript1.5" | "javascript1.6" | "javascript1.7" => return true,
+                            _ => {}
+                        },
+                        _ => {}
+                    }
+                }
+
+                let default_attributes = if namespace == Namespace::HTML {
+                    &HTML_DEFAULT_ATTRIBUTES
+                } else {
+                    &SVG_DEFAULT_ATTRIBUTES
+                };
+
+                let attributes = match default_attributes.get(tag_name) {
+                    Some(element) => element,
+                    None => return false,
+                };
```

Comment:
So you could write the code like this `let attributes = default_attributes.get(tag_name)?`